src/tools/idlc/src/descriptor.c

```diff
@@ -177,6 +177,11 @@ stash_opcode(
   struct instruction inst = { OPCODE, { .opcode = { .code = code, .order = order } } };
   const struct alignment *alignment = NULL;
 
+  if (code | DDS_OP_FLAG_EXT) {
```

Comment:
`code | DDS_OP_FLAG_EXT` is always true. From the looks of it, the intent was `code & DDS_OP_FLAG_EXT`.

Reply:
Yes, that was definitely wrong! I've fixed this and another issue in the rules to set NO_OPTIMIZE in commit 9c53626, and I've also added a test-case for this flag.